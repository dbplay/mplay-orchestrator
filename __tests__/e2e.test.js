const micro = require('micro')
const request = require('request-promise')
const listen = require('test-listen')
const handler = require('../index')

describe('mplay-orchestrator', () => {

    let service;
    let url;

    beforeAll(async () => {
        service = micro(handler)
        url = await listen(service)
    })

    afterAll(async () => {
        await request.get({ url: url + '/clean' })
        service.close()
})

it('can init send command and clean', async () => {
    jest.setTimeout(30 * 1000)
    const bodyInit = { version: '4.0' }
    const initRes = await request.post({ url: url + '/init', body: bodyInit, json: true })
    const id = initRes.id;
    const insertCommand = { id, command: 'db.test.insert({v: 4})'}
    const insertResult = await request.post({ url: url + '/command', body: insertCommand, json: true })
    expect(insertResult).toMatchSnapshot()

    const readCommand = { id, command: 'db.test.find()'}
    const readResult = await request.post({ url: url + '/command', body: readCommand, json: true })
    expect(readResult.status).toEqual('SUCCESS')
    expect(readResult.out).toMatch(/"v" : 4/)
})
})