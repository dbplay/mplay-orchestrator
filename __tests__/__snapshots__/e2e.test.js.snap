// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mplay-orchestrator can init send command and clean 1`] = `
Object {
  "out": "WriteResult({ \\"nInserted\\" : 1 })",
  "status": "SUCCESS",
}
`;
